pallets/loans/src/entities/pricing/external.rs

```diff
@@ -163,10 +169,26 @@ impl<T: Config> ExternalActivePricing<T> {
 		pool_id: T::PoolId,
 		maturity: Seconds,
 	) -> Result<T::Balance, DispatchError> {
-		Ok(match T::PriceRegistry::get(&self.info.price_id, &pool_id) {
-			Ok(data) => data.0,
-			Err(_) => self.linear_accrual_price(maturity)?,
-		})
+		self.current_price_inner(
+			maturity,
+			T::PriceRegistry::get(&self.info.price_id, &pool_id).ok(),
+		)
+	}
+
+	fn current_price_inner(
+		&self,
+		maturity: Seconds,
+		oracle: Option<PriceOf<T>>,
+	) -> Result<T::Balance, DispatchError> {
+		if let Some((oracle_price, oracle_provided_at)) = oracle {
+			self.linear_accrual_price(maturity, oracle_price, oracle_provided_at.into_seconds())
```

Comment:
Here, if the price comes from `T::PriceRegistry::get(&self.info.price_id, &pool_id).ok()` we're converting milliseconds to seconds. If the price comes from the cache, that was previously converted to seconds, we're converting to seconds again, which means divide by `1000`. Maybe this is related to the test issue you're experiencing.

Definitely, we need typed times to avoid this issue...

Reply:
Good catch!